hq/app/schedule/unrecognised/IamUnrecognisedUserJob.scala

```diff
@@ -1,33 +1,71 @@
+import com.gu.anghammarad.models.{AwsAccount => TargetAccount}
 import com.gu.janus.model.{ACL, AwsAccount, JanusData, SupportACL}
-import model.{CredentialReportDisplay, CronSchedule, HumanUser, AwsAccount => Account}
+import config.Config.getAnghammaradSNSTopicArn
+import model.{CronSchedule, VulnerableUser, AwsAccount => Account}
 import org.joda.time.Seconds
-import play.api.Logging
-import schedule.unrecognised.IamUnrecognisedUsers.{filterUnrecognisedIamUsers, getHumanUsers, getJanusUsernames}
+import play.api.{Configuration, Logging}
+import schedule.IamMessages.FormerStaff.disabledUsersMessage
+import schedule.IamMessages.disabledUsersSubject
+import schedule.Notifier.{notification, send}
+import schedule.unrecognised.IamUnrecognisedUsers.{filterUnrecognisedIamUsers, getJanusUsernames}
+import schedule.vulnerable.IamDisableAccessKeys.disableAccessKeys
+import schedule.vulnerable.IamRemovePassword.removePasswords
 import schedule.{CronSchedules, JobRunner}
 import services.CacheService
-import utils.attempt.FailedAttempt
+import utils.attempt.Attempt
 
-class IamUnrecognisedUserJob(cacheService: CacheService) extends JobRunner with Logging {
+import scala.concurrent.ExecutionContext
+
+class IamUnrecognisedUserJob(cacheService: CacheService, snsClient: AmazonSNSAsync, iamClients: AwsClients[AmazonIdentityManagementAsync], config: Configuration)(implicit val executionContext: ExecutionContext) extends JobRunner with Logging {
   override val id: String = "unrecognised-iam-users"
   override val description: String = "Check for and remove unrecognised human IAM users"
   override val cronSchedule: CronSchedule = CronSchedules.everyWeekDay
+  private val topicArn: Option[String] = getAnghammaradSNSTopicArn(config)
 
   def run(testMode: Boolean): Unit = {
     if (testMode) {
       logger.info(s"Skipping scheduled $id job as it is not enabled")
     } else {
       logger.info(s"Running scheduled job: $description")
     }
-  }
 
-  val dummyJanusData: JanusData = ???
-  val janusUsers: List[String] = getJanusUsernames(dummyJanusData)
-  val credsReport = cacheService.getAllCredentials
-  val iamHumanUsers: List[HumanUser] = getHumanUsers(credsReport)
-  val unrecognisedIamUsers: List[HumanUser] = filterUnrecognisedIamUsers(iamHumanUsers, "name", janusUsers)
+    val dummyJanusData = JanusData(
+      Set(AwsAccount("Deploy Tools", "deployTools")),
+      ACL(Map("firstName.secondName" -> Set.empty)),
+      ACL(Map.empty),
+      SupportACL(Map.empty, Set.empty, Seconds.ZERO),
+      None
+    )
+
+    val janusUsers: Seq[String] = getJanusUsernames(dummyJanusData)
+    val allCredsReports = cacheService.getAllCredentials
+
+    allCredsReports.map { case (account, eitherCredsReportOrFailure) =>
+      eitherCredsReportOrFailure.map { credsReport =>
+        val humanUsers = credsReport.humanUsers
+        val unrecognisedIamUsers: Seq[VulnerableUser] = filterUnrecognisedIamUsers(humanUsers, "name", janusUsers)
```

Comment:
The interface for `filterUnrecognisedIamUsers` opens up the possibility for the method to be called with different `targetKey` values. Is that the case? If not I'm wondering if it should be a constant in `IamUnrecognisedUsers`

Reply:
Absolutely, we discussed that but didn't go back and change it. Maybe we could address in #309 where it's introduced